be/src/exprs/expr_context.cpp

```diff
@@ -42,9 +42,7 @@ ExprContext::ExprContext(Expr* root)
 
 ExprContext::~ExprContext() {
     // DCHECK(!_prepared || _closed) << ". expr context address = " << this;
-    if (_prepared) {
-        close(_runtime_state);
-    }
+    close(_runtime_state);
```

Comment:
one quick question, close will be called twice?
One is by the users one is by the dtor.

Reply:
yes, all of our close method is safety to close multitimes.
```
void ExprContext::close(RuntimeState* state) {
    bool expected = false;
    if (!_closed.compare_exchange_strong(expected, true)) {
        return;
    }
```

```
ExecNode::~ExecNode() {
    if (runtime_state() != nullptr) {
        ExecNode::close(_runtime_state);
    }
}
```

Reply:
I understand. But seems not a good design.